internal/osutils/autostart/autostart_lin.go

```diff
@@ -2,17 +2,127 @@
 
 package autostart
 
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/gobuffalo/packr"
+	"github.com/mitchellh/go-homedir"
+)
+
 func (a *App) Enable() error {
-	panic("Not implemented")
+	appPath, err := ensuredAppPath()
+	if err != nil {
+		return errs.Wrap(err, "Could not ensure application file is present")
+	}
+
+	enabled, err := a.IsEnabled()
+	if err != nil {
+		return errs.Wrap(err, "Could not check if app autostart is enabled")
+	}
+	if enabled {
+		return nil
+	}
+
+	dir, err := dirPath(autostartDir)
+	if err != nil {
+		return errs.Wrap(err, "Could not get autostart file")
+	}
+	path := filepath.Join(dir, launchFile)
+
+	if err = os.Symlink(appPath, path); err != nil {
+		return errs.Wrap(err, "Could not create symlink")
+	}
 	return nil
 }
 
 func (a *App) Disable() error {
-	panic("Not implemented")
-	return nil
+	enabled, err := a.IsEnabled()
+	if err != nil {
+		return errs.Wrap(err, "Could not check if app autostart is enabled")
+	}
+	if !enabled {
+		return nil
+	}
+
+	dir, err := dirPath(autostartDir)
+	if err != nil {
+		return errs.Wrap(err, "Could not get autostart file")
+	}
+	path := filepath.Join(dir, launchFile)
+
+	return os.Remove(path)
 }
 
 func (a *App) IsEnabled() (bool, error) {
-	panic("Not implemented")
-	return false, nil
+	dir, err := dirPath(autostartDir)
+	if err != nil {
+		return false, errs.Wrap(err, "Could not get autostart file")
+	}
+	path := filepath.Join(dir, launchFile)
+
+	return fileutils.FileExists(path), nil
+}
+
+const (
+	applicationDir = ".local/share/applications"
+	autostartDir   = ".config/autostart"
+	iconsDir       = ".local/share/icons/hicolor/scalable/apps"
+	launchFile     = "state-tray.desktop"
+	iconFileState  = "state-tray.svg"
+	iconFileBase   = "icon.svg"
+)
+
+func dirPath(dir string) (string, error) {
+	homeDir, err := homedir.Dir()
+	if err != nil {
+		return "", errs.Wrap(err, "Could not get home directory")
+	}
+	return filepath.Join(homeDir, dir), nil
+}
+
+func ensuredAppPath() (string, error) {
```

Comment:
`ensuredAppPath` does not convey to me what it does. Please name this something more appropriate or add a comment.